File: Task3/app/src/main/java/com/example/task/ApiClient.java
package com.example.task;

import java.util.List;

import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.http.GET;

public class ApiClient {
    private static final String BASE_URL = "https://jsonplaceholder.typicode.com/";

    private static Retrofit retrofit = new Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build();

    public interface ApiService {
        @GET("photos")
        Call<List<MyDataModel>> getMyData();
    }

    public static ApiService getApiService() {
        return retrofit.create(ApiService.class);
    }
}

File: Task2/app/src/main/java/com/example/task2/ApiClient.kt
package com.example.task2

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET

object ApiClient {
    private const val BASE_URL = "https://jsonplaceholder.typicode.com/photos/"

    private val retrofit: Retrofit = Retrofit.Builder()
        .baseUrl(BASE_URL)
        .addConverterFactory(GsonConverterFactory.create())
        .build()

    interface ApiService {
        @GET("photos")
        suspend fun getMyData(): List<MyDataModel>
    }

    suspend fun fetchMyData(): List<MyDataModel> {
        val apiService = retrofit.create(ApiService::class.java)
        return apiService.getMyData()
    }
}
